<!doctype html>
<html>
<head>
  <title>slidewinder community docs</title>
  <link rel="stylesheet" type="text/css" href="../style.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/default.min.css">
</head>
<body>

  <div class="container">
    <section class="content">

        <p><a href="index.html">Back to table of contents</a></p>
        <h2 id="rna-seq-analysis">RNA-seq analysis</h2>
<p>In this tutorial we will learn the basic steps involved in an RNA-seq analysis.</p>
<p>Specifically we will focus on two major steps:</p>
<ol>
<li>quantifying transcript expression</li>
<li>differential expression analysis</li>
</ol>
<h3 id="initial-setup">Initial setup</h3>
<p>You should have a folder on your desktop called <code>RNA</code>.</p>
<p>It contains the data for this workshop. The first thing we will do is organise the files in this directory.</p>
<p>Open up a terminal. Now copy and paste the following commands into the terminal:</p>
<pre><code class="lang-bash">cd ~/RNA
mkdir analysis data plots software
mkdir data/salmon data/metadata data/reads data/reference
mv Arabidopsis* data/reference
mv Salmon* software
mv SRR* data/reads
</code></pre>
<h3 id="the-dataset">The dataset</h3>
<p>We will use data from a study uploaded to ArrayExpress in February 2016. It looked at expression in <em>Arabidopsis thaliana</em> plants that had their <a href="https://en.wikipedia.org/wiki/Pyridoxine_5%27-phosphate_oxidase">PMP/PNP oxidase</a> PDX3 genes knocked out, compared to wild-type plants with the gene intact.</p>
<p>The experiment is <a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-77428/">described by the authors</a> as follows:</p>
<blockquote>
<p>Arabidopsis mutants in the PMP/PNP oxidase PDX3 show abberant growth and development.RNA sequencing reveals strong induction of stress-related genes in pdx3, particularly those associated with biotic stress. Whole rosettes (21- days old) of three biological replicates of wild-type and pdx3-3 (SALK_054167C) and pdx3-4 (GK-260E03) mutants, all ecotype Col-0, were sequenced and a differential gene expression analysis was performed.</p>
</blockquote>
<p>Some things here may be unfamiliar to you:</p>
<ul>
<li><em>Arabidopsis thaliana</em> is the most well studied model plant species. It has a relatively small genome, a 6-week life cycle, and can be easily mutated, transformed and manipulated. It is usually called simply &#39;arabidopsis&#39;.</li>
<li><code>pdx3-3</code> and <code>pdx3-4</code> refer to mutant plant lines in which the <code>pdx3</code> gene is interrupted with a <a href="https://en.wikipedia.org/wiki/Transfer_DNA">TDNA insertion</a>.</li>
<li><code>SALK_054167C</code> and <code>GK-260E03</code> are the database identifiers of mutants ordered from a <a href="http://signal.salk.edu/tabout.html">mutant seed collection</a>.</li>
</ul>
<p><img src="../assets/Arabidopsis_thaliana.jpg" alt="Arabidopsis thaliana"></p>
<blockquote>
<p><em>Arabidopsis thaliana</em></p>
</blockquote>
<h3 id="our-analysis">Our analysis</h3>
<p>We will re-analyse the raw read data from the experiment described above.</p>
<p>Specifically we will:</p>
<ul>
<li>quantify expression for each sample</li>
<li>do some basic quality control on the resulting expression estimates</li>
<li>run a differential expression anaysis to identify genes expressed differently in the mutants compared to the wild-type plants</li>
</ul>

        <div class="navlinks">
          <a class="nextp button" href="01_quantifying.html">OK, next! &raquo;</a>
        </div>
    </section>
  </div>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
</body>
</html>
